app/models/miq_filter.rb

```diff
@@ -174,8 +174,12 @@ def self.apply_belongsto_filters(inputs, bfilters)
           end
           break if filtered.last == p
         else
-          vcmeta = belongsto2object(tag)
+          vcmeta_list = belongsto2object_list(tag)
+          vcmeta_list.pop if vcmeta_list.last.kind_of?(Host)
+          vcmeta = vcmeta_list.last
```

Comment:
@jrafanie 

Let assume Group has stored stored filter Host1, 
but it is not reflected UI:


![111](https://cloud.githubusercontent.com/assets/14937244/13106961/68c85b3c-d56b-11e5-9acc-d4de1d15a254.png)

it fixes this change https://github.com/lpichler/manageiq/commit/dada4c655e983582cf68dd874a9ba3a85e483d61#diff-6ceef9efd67c362225272c1ae94b6c38R136 in MiqFilter.belongsto2object_list
**but not completelly**,  unselected Hosts disappeared:

![333](https://cloud.githubusercontent.com/assets/14937244/13106974/784a4e4e-d56b-11e5-9ea2-4ca3c5ae7a02.png)


and it solves this fix, (what are you asking for)
https://github.com/lpichler/manageiq/commit/dada4c655e983582cf68dd874a9ba3a85e483d61#diff-6ceef9efd67c362225272c1ae94b6c38R180 :
![22](https://cloud.githubusercontent.com/assets/14937244/13106966/6dd7d9ea-d56b-11e5-9166-f193d91604b5.png)

It is because belongsto2object with parameter /belongsto/ExtManagementSystem|X1/EmsFolder|X2/EmsFolder|X2/**EmsFolder|host**/Host|HOST1
returns Host object now and before it was returning EmsFolder object for pattern marked with bold
so MiqFitler.apply_belongs_to_filters needs to react in same way as before.

Is it clearer now ?